go/libraries/doltcore/sqle/tables.go

```diff
@@ -608,12 +609,55 @@ func (t *WritableDoltTable) AutoIncrementSetter(ctx *sql.Context) sql.AutoIncrem
 
 // GetAutoIncrementValue gets the last AUTO_INCREMENT value
 func (t *WritableDoltTable) GetAutoIncrementValue(ctx *sql.Context) (interface{}, error) {
+	sess := DSessFromSess(ctx.Session)
+
 	if !t.autoIncCol.AutoIncrement {
 		return nil, sql.ErrNoAutoIncrementCol
 	}
+
+	autoIncTracker, foundAit := sess.GetDoltDbAutoIncrementTracker(t.db.Name())
+	if !foundAit || autoIncTracker == nil {
+		return t.getAutoIncrementValue(ctx)
+	}
+
+	stored, err := t.getAutoIncrementValue(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ok, err := autoIncTracker.Request(t.tableName, stored)
```

Comment:
Just return a best guess here, with the understanding that the table editor will replace it with the true value before committing.